### PR TITLE
Update Feature Detection in cm-api-shim.js

### PR DESCRIPTION
**Pull request by jdoerrie**
This change updates the feature detection used in cm-api-shim.js.

### CODE REVIEW

File: cm-api-shim.js [hunk 1]
```diff
@@ -18,7 +18,8 @@
 (function() {
   // Return early if CM API is not available or the new features are already
   // present.
-  if (!navigator.credentials || navigator.credentials.preventSilentAccess)
+  if (!window.PasswordCredential || !window.FederatedCredential || 
+      navigator.credentials.preventSilentAccess)
```
Comment (jdoerrie) [inline cm-api-shim.js]: Potentially we need to update line 55 as well, as it is theoretically possible that a browser implements the old CM API in conjunction with the WebAuthN API (this wouldn't be the case for Chrome though). What is your opinion here?

Comment (agektmr) [reply cm-api-shim.js]: I think it's possible that there will be a browser which implements `PasswordCredential` but `FederatedCredential`. I don't think we need to change this line.

Comment (agektmr) [reply cm-api-shim.js]: I think adding checks for `PasswordCredential` and `FederatedCredential` in line 69 and 78 would make sense.

**Review by agektmr — CHANGES_REQUESTED** [review]

File: cm-api-shim.js (hunk 1 above)
Comment (jdoerrie) [reply cm-api-shim.js]: Done

File: cm-api-shim.js [hunk 2]
```diff
@@ -51,41 +50,50 @@
...
   }
 
-  // Polyfill for navigator.credentials.create.
-  navigator.credentials.create = function(options) {
-    var error_msg = "NotSupportedError: Only 'password' and 'federated'" +
-        'credential types are currently supported.';
+  // Polyfill for Credential Management API's navigator.credentials.create.
+  // Other APIs might define this method already, in which case we need to cache
+  // the original version.
+  if (navigator.credentials.create) {
```
Comment (jdoerrie) [inline cm-api-shim.js:56]: I added this logic to be able to dispatch to an already existing version of navigator.credentials.create if applicable. WDYT?

Comment (agektmr) [reply cm-api-shim.js:56]: I like it. LGTM

**Review by agektmr — APPROVED** [review]

**Comment by jdoerrie** [comment]
Thanks! I merged the PR and created a new release and tag, so that bower can pick up the new changes.

**Pull request merged**